cmd/bnsd/scenarios/multisig_test.go

```diff
@@ -63,4 +63,55 @@ func TestMultisigCanPayFees(t *testing.T) {
 		t.Fatalf("cannot broadcast new revenue transaction: %s", err)
 	}
 
+	multisigID := resp.Response.DeliverTx.Data
+	t.Logf("multisig created with id: %x", multisigID)
+
+	// Final step is to put coins on the newly created multisig account so
+	// that it can be used to pay with by the members.
+	multisigAddr := multisig.MultiSigCondition(multisigID).Address()
+	bnsdtest.SeedAccountWithTokens(t, env, multisigAddr)
+
+	// Test setup is done, now the actual test.
+	//
+	// User Carl does not have any funds but multisig does. Ensure that
+	// Carl using multisig can execute operations that will collect fee
+	// from the multisig account.
+	// Let us create an another multisig instance as this operation
+	// requires a fee payment.
+
+	anotherNewMultisigTx := &bnsdApp.Tx{
+		// When using a multisig, transaction must be configured
+		// additionally. Decorators are looking into extra fields,
+		// never into the message content.
+		Multisig: [][]byte{multisigID},
+
+		Sum: &bnsdApp.Tx_CreateContractMsg{
+			CreateContractMsg: &multisig.CreateContractMsg{
+				Metadata: &weave.Metadata{Schema: 1},
+				Participants: []*multisig.Participant{
+					&multisig.Participant{
+						Signature: carl.PublicKey().Address(),
+						Weight:    1,
+					},
+				},
+				ActivationThreshold: 1,
+				AdminThreshold:      1,
+			},
+		},
+	}
+
+	anotherNewMultisigTx.Fee(multisigAddr, coin.NewCoin(1, 1, "IOV"))
+
+	carlNonce := client.NewNonce(env.Client, carl.PublicKey().Address())
+	if seq, err := carlNonce.Next(); err != nil {
+		t.Fatalf("cannot acquire carl nonce sequence: %s", err)
+	} else if err := client.SignTx(anotherNewMultisigTx, carl, env.ChainID, seq); err != nil {
```

Comment:
Why use `else if` here if previous if exits on failure?
I already find lines like `if err := client.SignTx(anotherNewMultisigTx, carl, env.ChainID, seq); err != nil` a bit annoying to read when they include the principle action we need to test `client.SignTx` mixed in with the error check.
And using `else if` just to avoid a newline seems again a readability loss.

Honestly, I think this is equivalent and most readable. But if you like the explicit t.Fatalf, feel free to use it. Using odd constructs not needed by code (eg. else if, switch) just to avoid a few newlines or `{}` seems to hurt readability for a minimal gain in less typing.

```
seq, err := carlNonce.Next()
assert.Nil(t, err)
err = client.SignTx(anotherNewMultisigTx, carl, env.ChainID, seq) 
assert.Nil(t, err)
```

Looking again, I see the `else if` is needed so the `seq` from the first if statement is in scope.
I would really request you and Alex stop using such convoluted scoping, and write more straight-forward checks in tests.

Reply:
I agree that those `if/else` are not great. What do you think about https://github.com/iov-one/weave/pull/650/commits/9a0323a0261e77db80aebb1e2a95fe26e381dbce ?

I do like explicit `t.Fatalf` because every time `assert.Nil` fails, it is producing a useless error. I think it is a huge advantage of `t.Fatalf` that you can describe what operation failed. It seems like writing an obvious message when writing a test, but it is a very helpful when reading it some other time. Especially when the test is failing and you wonder _"ok, this cannot be nil, but **why**?"_

Reply:
Much better. Thank you.

I understand and respect your opinon wrt. `assert.Nil`. I only offer it as a candidate if you want to save key-strokes. The changes you made in the above-mentioned commit solve my complains and look a whole lot more readable to me. Thank you.